Shopify/quilt
# Reduce usage of loom in favor of calling underlying tools directly

## Description

Loom was a neat idea but it is without an owner that can give it the love it deserves. To de-risk quilt, let's move back to using underlying tools directly.

This PR moves linting, test running and type-checking behaviour back to calling the underlying tools directly and removes configuration for these behaviours from loom. Our dependency on `@shopify/loom-plugin-build-library` is removed and replaced with simplified versions in the config folder that do not seek to configure `typescript` and `jest`.

- `yarn lint` calls `eslint` and `prettier` directly. `loom lint` now does nothing.
- `yarn test` calls `jest` directly, and is configured through the root level `jest.config.js` file. `loom test` now does nothing.
- `yarn type-check` calls `tsc` directly. `loom type-check` now does nothing.

I've opted to keep cache data in a `.loom` directory for now, that can be cleared up once we build is migrated.


At this point `loom build` is the only command that has any effect.


As we now call jest directly, you'll need to call `yarn test --watch` if you want to create a watching test watcher.


### To test

- Ensure that `yarn lint` continues to test content with eslint and prettier
- Ensure that `yarn test` runs all jest tests. Note that total test count is the same as before - 3826 tests spread over 219 test suites.
- Ensure that `yarn type-check` runs a type check.


Compare build output to previous builds:

- Check out the main branch of quilt in your `~/projects/quilt` directory and run `yarn`, `yarn clean && yarn build` to produce a clean build
- Check out this branch of quilt in your `~/src/github.com/Shopify/quilt` directory and run `yarn`, `yarn clean && yarn build` to produce a clean build
- Run ` for PACKAGENAME in $(ls -1 packages); diff -ru ~/projects/quilt/packages/$PACKAGENAME/build packages/$PACKAGENAME/build -x '*.tsbuildinfo' -x '*.d.ts.map'` to compare the build folder output and note that it is identical
